### PR TITLE
Switch to explicitly whitelisting tasks for OSX

### PR DESCRIPTION
**Pull request by DRMacIver**
We'd like to run as little on OSX as possible, as the Travis
builds for it are very slow. Thus it makes sense to whitelist
rather than blacklist which jobs should run on it.

(Note: I'm not 100% sure this is the right syntax and will need to see what Travis does)

### CODE REVIEW

**Comment by DRMacIver** [comment]
Note: This probably supersedes #462

File: .travis.yml [hunk 1]
```diff
@@ -59,52 +59,13 @@ script:
...
-          env: TASK=check-examples2
-        - os: osx
-          env: TASK=check-examples3
-        - os: osx
-          env: TASK=check-coverage
-        - os: osx
-          env: TASK=check-format
-        - os: osx
-          env: TASK=lint
-    fast_finish: true
```
Comment (alexwlchan) [inline .travis.yml]: Did you mean to drop this line? I think it’s useful, particularly given how long a full Hypothesis build otherwise takes.

Comment (DRMacIver) [inline .travis.yml]: Yeah, I should have done this in a separate commit/PR, sorry. Removing it is fine though: The semantics of fast\_finish unfortunately make it more or less useless. It only causes the build to fast finish if the only remaining jobs are ones which are allowed to fail (which we don't have).

**Comment by DRMacIver** [comment]
Hmm. A thing I *don't* like about this is that it now schedules all the OSX jobs at the end. This is bad given how long they take to actually spin up.

There doesn't seem to be a documented way to deliberately control the build order. :-( I'll have a tinker with this and see what I can do.

File: .travis.yml (hunk 1 above)
Comment (alexwlchan) [inline .travis.yml]: From [the Travis docs](https://docs.travis-ci.com/user/customizing-the-build/) (emphasis mine):

> Now, a build will finish **as soon as a job has failed**, or when the only jobs left allow failures.

That feels like it might be desirable? Given that a full run still takes 2½+ hours and there’s a limit to how many concurrent builds Travis will run, failing a build once the canary jobs have failed seems like it might be useful.

**Comment by alexwlchan** [comment]
Not sure if the test failure is spurious or not – I see a random seed being thrown about, so maybe worth looking into before I just re-run the job?

File: .travis.yml (hunk 1 above)
Comment (alexwlchan) [inline .travis.yml]: Actually, if that’s correct, I think we _do_ want to have this flag enabled, and keep the running order as is – then the slow OS X builds will only run if we’re sure there aren’t failures lurking elsewhere.

Comment (DRMacIver) [inline .travis.yml]: Hmm. I'm happy to add it back in, but empirically that doesn't seem to be what happens at all.

**Comment by DRMacIver** [comment]
I think that failing test is another one which became flaky with recent changes, so I'm not going to worry about it until I see it fail again ahead of current master.
